### ISSUE TITLE
LLM selection UI is not accessible (name and role)

### ISSUE THREAD

**Issue by derekjackson-das** [danny-avila/LibreChat#3473]
**Issue Description:** The menu button and menu list are not accessible for a few reasons and there are a few WCAG SC issues in this element. Primarily, there is no accessible, programmatic name for the element and the element has no role (currently just identified as "generic" and so naming issues will not be found by automated checkers). There are also conflicting aria attributes in the UI (aria-haspopup set to dialog, but it opens a dialog with menulist which must be part of a menu/menubar composite widget) and lack of indication of the state (no indication when an item in the list is selected.) To make the UI of the element accessible the role and name should be identified with a clear composite widget role to be used as well as any [accessibility descendants](https://w3c.github.io/aria/#dfn-accessibility-descendant). 

**WCAG Criteria:** [3.3.2 Labels or Instructions - A](https://www.w3.org/WAI/WCAG21/Understanding/labels-or-instructions), [4.1.2 Name, Role, Value](https://www.w3.org/WAI/WCAG22/Understanding/name-role-value.html), [Label in name 2.5.3](https://www.w3.org/WAI/WCAG22/Understanding/label-in-name.html)

**Screenshot or Snippet:** [Link to Screenshot](https://share.zight.com/Apum4koG)

**Notes:** A name for this field could be "LLM Model" and then the value would be the LLM. selected Then the Value will be described and the label name, eg. "Anthropic Claude, LLM Model".  A useful pattern to consider for the UI of this element is the [listbox](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-select-only/) or single column [datagrid](https://www.w3.org/WAI/ARIA/apg/patterns/grid/examples/data-grids/) to accommodate nested button and actions.

**Comment by danny-avila** [comment]
Is now resolved:

![image](https://github.com/user-attachments/assets/cc71abfa-d0bb-409a-a187-154550faeec8)
